src/components/v5/shared/Modal/partials/TokensModal/TokensModal.tsx

```diff
@@ -0,0 +1,125 @@
+import React, { FC } from 'react';
+
+import Modal from '../../Modal';
+import Numeral from '~shared/Numeral';
+import { SpinnerLoader } from '~shared/Preloaders';
+import Button from '~v5/shared/Button';
+import FormFormattedInput from '~v5/common/Fields/InputBase/FormFormattedInput';
+import { formatText } from '~utils/intl';
+import { ActionForm } from '~shared/Fields';
+import { TokensModalProps } from './types';
+import { useTokensModal } from './hooks';
+import TokenIcon from '~shared/TokenIcon';
+
+const displayName = 'v5.Modal.partials.TokensModal';
+
+const TokensModal: FC<TokensModalProps> = ({ type, onClose, ...props }) => {
+  const {
+    validationSchema,
+    actionType,
+    tokenBalanceData,
+    tokenDecimals,
+    nativeToken,
+    transform,
+    tokenSymbol,
+    pollActiveTokenBalance,
+    tokenBalanceInEthers,
+    loading,
+  } = useTokensModal(type);
+
+  return (
+    <Modal {...props} onClose={onClose}>
+      <ActionForm
+        actionType={actionType}
+        defaultValues={{ amount: '0' }}
+        validationSchema={validationSchema}
+        transform={transform}
+        onSuccess={(_, { reset }) => {
+          pollActiveTokenBalance();
+          reset();
+          onClose();
+        }}
+      >
+        {({ setValue }) => (
+          <>
+            <h4 className="heading-5 mb-1.5">
+              {formatText({ id: `tokensModal.${type}.title` })}
+            </h4>
+            <p className="text-md text-gray-600 mb-6">
+              {formatText({ id: `tokensModal.${type}.description` })}
+            </p>
+            <div className="mb-8">
+              <div className="flex items-center justify-between gap-2 mb-1">
+                <p className="text-1">
+                  {formatText({ id: `tokensModal.${type}.input` })}
+                </p>
+                <span className="text-sm text-gray-600 flex items-center gap-1">
+                  {formatText({ id: 'tokensModal.balance' })}
+                  {loading ? (
+                    <SpinnerLoader appearance={{ size: 'small' }} />
+                  ) : (
+                    <Numeral
+                      value={tokenBalanceData || 0}
+                      decimals={tokenDecimals}
+                      suffix={tokenSymbol}
+                    />
+                  )}
+                </span>
+              </div>
+              <FormFormattedInput
+                name="amount"
+                customPrefix={
+                  <>
+                    {nativeToken && (
+                      <div className="absolute top-0 left-0 px-3.5 py-3 h-full flex items-center justify-center">
```

Comment:
Doesn't it go under the content of the input? I didn't see any padding added to it for such case